sig-and-wg/sig-core/proposals/installer.md

```diff
@@ -0,0 +1,42 @@
+# Kyma Installer - dynamic installation 
+
+Created on 2018-08-23 by Krysian Cieślik (@crabtree) and Jakub Kabza (@jakkab).
+This document is a part of the [Kyma modularization proposal](./modularization.md).
+
+## Status
+
+Proposed on 2018-08-23.
+
+## Motivation
+
+Kyma installation is a complex, time-consuming process, involving multiple components. The procedure is followed every day by a large number of developers who test their features, and thus should easily handle different scenarios without modifications. This is where the dynamic installation comes in handy, unifying, simplifying and automating the process.
+
+## Goal
+
+This proposal aims to enhance developer experience by:
+
+- extending installation customizability 
+- reducing the number of different installation methods
+- avoiding redundant modifications to the Installer component
+
+## Current solution
+
+Kyma installation:
+
+- consists of hard-coded steps
+- uses a variety of installation methods throughout the process, such as custom shell scripts, Helm, and kubectl commands
+- stores overrides in a fixed struct that requires code modifications every time new parameter is introduced
+
+All things considered, the installation process lacks coherence and leaves little room for customization, requiring constant maintenance at the same time.
+
+## Suggested solution
+
+- Installer is a component that handles installation, update, and deletion of Helm charts.
+- Installer no longer supports Bash scripts.
```

Comment:
does that mean you will implement a kind of CLI now, right?

Reply:
what I meant is, if we do not do run.sh, what else then? kubectl apply?

Reply:
we are talking here about component installation (e.g. core, helm broker bundles) not running kyma locally (e.g. firing run.sh script). Installer handles kyma components installation (e.g. core, istio, dex, remote-envs and so on), run.sh handles setting up minikube, rbac, kyma prerequisites, installer itself.

Reply:
so maybe additional point in the Goal section should be:
`Not a goal of this proposal: Usage of bash to trigger installation and all related preparation like for example generation of the Installation resource`

Reply:
@jakkab you kind of merged ignoring my feedback ;)

Reply:
@derberg Oops, sorry, I will fix it in a minute.